src/main/java/org/sonarlint/intellij/config/project/SearchProjectKeyDialog.kt

```diff
@@ -0,0 +1,179 @@
+    parent: Component,
+    private val lastSelectedProjectKey: String?,
+    private val projectsByKey: Map<String, ServerProject>,
+    private val isSonarCloud: Boolean
+) : DialogWrapper(
+    parent, false
+) {
+    private lateinit var projectList: JBList<ServerProject>
+    private lateinit var searchTextField: SearchTextField
+
+    init {
+        title = "Select " + (if (isSonarCloud) "SonarCloud" else "SonarQube") + " Project To Bind"
+        init()
+    }
+
+    override fun createCenterPanel() = panel {
+        row {
+            searchTextField = SearchTextField()
+            searchTextField.textEditor.emptyText.text = "Search by project key or name"
+            searchTextField.addDocumentListener(object : DocumentAdapter() {
+                override fun textChanged(e: DocumentEvent) {
+                    updateProjectsInList()
+                }
+            })
+            searchTextField().growPolicy(GrowPolicy.MEDIUM_TEXT)
+        }
+        row {
+            projectList = createProjectList()
+            scrollPane(projectList)
+            updateProjectsInList()
+        }
+    }
+
+    private fun updateOk(): Boolean {
+        val valid = selectedProjectKey != null
+        myOKAction.isEnabled = valid
+        return valid
+    }
+
+    val selectedProjectKey: String?
+        get() {
+            val project = projectList.selectedValue
+            return project?.key
+        }
+
+    private fun createProjectList(): JBList<ServerProject> {
+        val projectList = JBList<ServerProject>(DefaultListModel<ServerProject>())
+        val emptyText = StringBuilder("No projects found")
+        if (projectsByKey.isEmpty()) {
+            emptyText.append(" for the selected connection")
+        }
+        projectList.setEmptyText(emptyText.toString())
+        projectList.cellRenderer = ProjectListRenderer()
+        projectList.addListSelectionListener(ProjectItemListener())
+        projectList.addMouseListener(ProjectMouseListener())
+        projectList.selectionMode = ListSelectionModel.SINGLE_SELECTION
+        projectList.visibleRowCount = 10
+        projectList.border = IdeBorderFactory.createBorder()
+        projectList.cellRenderer = ProjectListRenderer()
+        return projectList
+    }
+
+    private fun updateProjectsInList() {
+        val filterText = searchTextField.text
+        val selection: ServerProject? = projectList.selectedValue
+        val model = (projectList.model as? DefaultListModel)
+        model!!.clear()
+        val projectComparator = java.util.Comparator { o1: ServerProject, o2: ServerProject ->
+            val c1 = o1.name.compareTo(o2.name, ignoreCase = true)
+            if (c1 != 0) {
+                return@Comparator c1
+            }
+            o1.key.compareTo(o2.key, ignoreCase = true)
+        }
+        val sortedProjects = projectsByKey.values.stream()
+            .sorted(projectComparator)
+            .collect(Collectors.toList())
```

Comment:
I think you can replace those lines with:
`val sortedProjects = projectsByKey.values.sortedWith(compareBy({ it.name.toLowerCase() }, { it.key.toLowerCase() }))`